src/lib/components/Cytoscape.react.js

```diff
@@ -798,7 +810,84 @@ Cytoscape.propTypes = {
+     *     - `itemTextShadowColor` (string): The text shadow colour of the command's content.
+     *     - `zIndex` (number): The z-index of the ui div.
+     *     - `atMouse` (bool): Draw menu at mouse position.
+     *
+     * 2. The format array inside the `commands` dictionary which determines how information is passed back.
+     * In the format array, strings can be entered to determine which properties of the cytoscape json object are selected.
+     * Alternatively, instead of a string, a dictionary can be used to select properties more specifically. The properties of such an object are outlined below:
+     *     - `key` (string): Specifies the property to be selected in the json object.
+     *     - `props` (array): This array specifies the properties of the key which are selected.
+     *     - `filter` (dictionary): This property is only to be applied where the value of the `key` is an array of objects. This object specifies the properties and values that the object in the array must have to not be removed from the array.
+     * For an example of the capabilities of this system, please reference the ctxmenu demo.
+     */
+    ctxmenu: PropTypes.arrayOf(
+        PropTypes.shape({
+            selector: PropTypes.string,
+            commands: PropTypes.arrayOf(
+                PropTypes.shape({
+                    id: PropTypes.string.isRequired,
+                    content: PropTypes.string.isRequired,
+                    contentStyle: PropTypes.object,
+                    fillColor: PropTypes.string,
+                    enabled: PropTypes.bool,
+                    format: PropTypes.array
+                })
+            ),
+            menuRadius: PropTypes.number,
+            fillColor: PropTypes.string,
+            activeFillColor: PropTypes.string,
+            activePadding: PropTypes.number,
+            indicatorSize: PropTypes.number,
+            separatorWidth: PropTypes.number,
+            spotlightPadding: PropTypes.number,
+            minSpotlightRadius: PropTypes.number,
+            maxSpotlightRadius: PropTypes.number,
+            openMenuEvents: PropTypes.string,
+            itemColor: PropTypes.string,
+            itemTextShadowColor: PropTypes.string,
+            zIndex: PropTypes.number,
+            atMouse: PropTypes.bool
+        })
+    ),
+
+    /**
+     * The dictionary returned when a ctxmenu option is selected. Read-only.
+     *
+     *     1. The structure of the dictionary is as follows:
+     *         - `id` (string): User supplied string meant to identify the specific ctxmenu option triggered
+     *         - `timestamp` (number): Millisecond UNIX timestamp indicating the time the ctxmenu option was selected
+     *         - `data` (dictionary): Data dump containing information on element ctxmenu is triggered on. This data is fetched by cytoscape json function.
+     */
+    ctxmenuData: PropTypes.object
```

Comment:
If you know the structure, it's better to document it using `PropTypes.exact` rather than just text. Even for read-only props this improves the documentation, particularly when we build for languages other than Python.